Automattic/wp-calypso
# CPT: Prompt and permanently delete trashed posts

This pull request seeks to enable permanently deleting trashed posts. If a post is already trashed, the trash button should permanently delete the post. The user will be prompted to confirm this action before issuing the request.

__Implementation notes:__

I opted against property-specific selectors in f4951c41010c2ffab2c89ea96f5e6fb55c76f31b, since we can achieve most of the items below in our `getNormalizedPost` selector.

~~The individual property selectors may seem a bit excessive, and could be tedious to implement for every property. However, the one's included highlight a few benefits:~~

- ~~Consistency with naming conventions ([reference](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#naming-conventions))~~
- ~~Opportunity to customize logic, similar to [where we accept options for generating URLs of media items](https://github.com/Automattic/wp-calypso/blob/master/client/lib/media/utils.js#L31-L72)~~
- ~~Avoids need to test for presence of `post` before accessing nested properties~~
- ~~Avoids tying our components to shape of state structure if we refactor post object~~

Related discussion: https://github.com/Automattic/wp-calypso/pull/6082/files#r67393626

__Testing instructions:__

Ensure Mocha tests pass:

```
npm run test-client client/state/posts
```

Verify that when trashing a post from the [custom post type list screen](http://calypso.localhost:3000/types/post), that...
 - If the post is not yet trashed, the post is immediately removed and sent to trash (no prompt)
 - If the post is already trashed, you are prompted to permanently delete the post and only if you accept this prompt is the post removed (deleted)

Test live: https://calypso.live/?branch=update/cpt-permanent-delete
